pipeline-runner/R/gem2s-2-load_user_files.R

```diff
@@ -71,19 +72,38 @@ read_10x_files <- function(config, input_dir) {
       paste(sample_fpaths, collapse = " - ")
     )
 
-    counts <- Seurat::Read10X(sample_dir, gene.column = 1)
+    annot <- read.delim(annot_fpath, header = FALSE)
 
-    if (is(counts, "list")) {
-      slot <- "Gene Expression"
-      # questionable: grab first slot if no gene expression
-      if (!slot %in% names(counts)) slot <- names(counts)[1]
-      counts <- counts[[slot]]
+    if (ncol(annot) == 1 || annot[1, 2] == "Gene Expression") {
+      annot[, 2] <- annot[, 1]
     }
 
-    annot <- read.delim(annot_fpath, header = FALSE)
+    features_types <- extract_features_types(annot)
+
+    message("Features types is ", features_types, "for sample ", sample)
+
+    if (features_types == -1) {
+      annot[, c(1, 2)] <- annot[, c(2, 1)]
+      gene_column <- 2
+      features_types <- 1
+    }
```

Comment:
the previous if is imopssible to read for anyone without the seurat object, moreover I think it's very error prone. I'd try to make it more readable by either adding an explanation or naming the hardcoded values as constants (if it makes sense)

Reply:
This is complex because there's basically no alternative. It's a table with two columns and we need to reverse columns 1 and 2

Reply:
hey @ogibson, don't we prefer count matrices with ENS ID's as rownames? I understand that from the fact that gene_column defaults to 1. So why do you change it here?

Reply:
oooh, I see. Read10X does not look to the annot object, but reads it again from the original features file